### PR TITLE
Updating training images for konflux build

### PR DESCRIPTION
**Pull request by Fiona-Waters**
Related to [RHOAIENG-15034](https://issues.redhat.com/browse/RHOAIENG-15034)
## Description

The following errors have been encountered in Konflux builds for the CUDA and ROCm Training images. 
```
[check-container]             {
[check-container]                 "name": "RunAsNonRoot",
[check-container]                 "elapsed_time": 0,
[check-container]                 "description": "Checking if container runs as the root user because a container that does not specify a non-root user will fail the automatic certification, and will be subject to a manual review before the container can be approved for publication",
[check-container]                 "help": "Check RunAsNonRoot encountered an error. Please review the preflight.log file for more information.",
[check-container]                 "suggestion": "Indicate a specific USER in the dockerfile or containerfile",
[check-container]                 "knowledgebase_url": "https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html-single/red_hat_openshift_software_certification_policy_guide/index#assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction",
[check-container]                 "check_url": "https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html-single/red_hat_openshift_software_certification_policy_guide/index#assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction"
[check-container]             },
[check-container]             {
[check-container]                 "name": "HasModifiedFiles",
[check-container]                 "elapsed_time": 661417,
[check-container]                 "description": "Checks that no files installed via RPM in the base Red Hat layer have been modified",
[check-container]                 "help": "Check HasModifiedFiles encountered an error. Please review the preflight.log file for more information.",
[check-container]                 "suggestion": "Do not modify any files installed by RPM in the base Red Hat layer",
[check-container]                 "knowledgebase_url": "https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html-single/red_hat_openshift_software_certification_policy_guide/index#assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction",
[check-container]                 "check_url": "https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html-single/red_hat_openshift_software_certification_policy_guide/index#assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction"
[check-container]             }
```
 - I have added USER 0 so that it will run as root which should remove the RunAsRoot error.
 - I have updated the base image to include the python version which should alleviate the error around modifying files - although the version of the requests package needs to be updated - I am not sure if this will cause the same error.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: images/runtime/training/cuda/Dockerfile [hunk 1]
```diff
@@ -16,20 +16,11 @@ LABEL name="training:py311-cuda121-torch241" \
 COPY LICENSE.md /licenses/cuda-license.md
 
 # Set the working directory in the container
+USER 0
```
Comment (sutaakar) [inline images/runtime/training/cuda/Dockerfile:19]: Can you set back `USER 1001` same as it is done for ROCm?

Comment (sutaakar) [reply images/runtime/training/cuda/Dockerfile:19]: I mean in the end, after all the needed installations

Comment (Fiona-Waters) [reply images/runtime/training/cuda/Dockerfile:19]: Updated, thank you

File: images/runtime/training/rocm/Dockerfile [hunk 2]
```diff
@@ -16,20 +16,11 @@ LABEL name="training:py311-rocm61-torch241" \
...
-    && dnf update -y \
-    && dnf clean all
-
-RUN dnf remove -y python3-requests && \
-    pip install --no-cache-dir requests==2.32.3
+# upgrade requests package
+RUN pip install --no-cache-dir --upgrade requests==2.32.3
 
 # Install ROCm
 USER 0
```
Comment (astefanutti) [inline images/runtime/training/rocm/Dockerfile:26]: Should that be removed?

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Dockerfile:26]: As far as I remember there were issues and the requests package needed to be set specifically. Let me try removing it and I can pin it in the pipfile if necessary. Thanks

Comment (astefanutti) [reply images/runtime/training/rocm/Dockerfile:26]: I mean as it's added line 19, can this be removed from line 26?

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Dockerfile:26]: Sorry I misunderstood which line you were referring to. Yes this should be removed.

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by astefanutti — APPROVED** [review]

**Pull request merged**